app/src/lib/promise.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Wrap a promise in a minimum timeout, so that it only returns after both the
+ * timeout and promise have completed.
+ *
+ * This is ideal for scenarios where a promises may complete quickly, but the
+ * caller wants to introduce a minimum latency so that any dependent UI is
+ *
+ *
+ * @param action the promise work to track
+ * @param timeout the minimum time to wait before resolving the promise (in milliseconds)
+ */
+export function promiseWithMinimumTimeout<T>(
```

Comment:
@desktop/core @desktop/comrades this is probably the only non-UI change that needs a review as part of this PR. I'd love some input on it.

Reply:
I guess my main feedback would be add some unit tests? If this is something we intend on using going forward, it would be nice to know its been tested.

Reply:
@say25 good shout. I'll see what I can do to control `window.setTimeout` now that we're on Jest.

Reply:
@shiftkey https://jestjs.io/docs/en/timer-mocks.html 😃

Reply:
I've pushed 174cbb292 which covers the interesting scenarios to test for